.github/workflows/documentation.yml

```diff
@@ -0,0 +1,83 @@
+name: Build and deploy documentation
+
+on:
+  push:
+    branches: [ master ]
+  workflow_dispatch:
+  pull_request:
+
+jobs:
+  check-documentation:
+    if: github.event_name != 'push'
```

Comment:
any reason not to use `== 'pull_request'`?

Reply:
Yes. We still want to allow the `workflow_dispatch` (that is a manual trigger via the website) event to trigger this workflow.